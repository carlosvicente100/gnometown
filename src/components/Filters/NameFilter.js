import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const InputName = styled.input`
  width: -webkit-fill-available;
  height: 40px;
  font-size: 28px;
  margin: 20px;
`

const NameFilter = ({ onChange, current }) => {
  const handleChange = (event) => {
    event.preventDefault()
    onChange(event.target.value)
  }
  return (
    <div>
      <InputName
        data-testid="filter-name-input"
        type="text"
        onChange={handleChange}
        value={current}
        placeholder="Gnome Name"
      />
    </div>
  )
}

NameFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  current: PropTypes.string.isRequired
}

export default NameFilter
