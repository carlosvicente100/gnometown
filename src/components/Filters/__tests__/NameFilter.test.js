import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import NameFilter from '../NameFilter'

const dummyProps = {
  current: 'Dummy Name',
  onChange: jest.fn()
}

describe('NameFilter Component Suite', () => {
  test('Snapshot check', () => {
    const component = renderer.create(<NameFilter {...dummyProps} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('NameFilter Structure', () => {
  test('Should have an input for each list item - current', () => {
    const { getByTestId } = render(<NameFilter {...dummyProps} />)
    const nameInput = getByTestId('filter-name-input')

    expect(nameInput.getAttribute('value')).toEqual(dummyProps.current)
    expect(nameInput.getAttribute('placeholder')).toEqual('Gnome Name')
  })
})
describe('NameFilter Structure', () => {
  test('Should call onChange when change content - onChange', () => {
    const newInputValue = 'Kent C. Dodds'
    const { getByTestId } = render(<NameFilter {...dummyProps} />)
    const nameInput = getByTestId('filter-name-input')

    fireEvent.change(nameInput, { target: { value: newInputValue } })

    expect(dummyProps.onChange).toHaveBeenCalledTimes(1)
  })
})
