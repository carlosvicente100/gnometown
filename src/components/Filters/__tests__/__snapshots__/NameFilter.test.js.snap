// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameFilter Component Suite Snapshot check 1`] = `
.c0 {
  width: -webkit-fill-available;
  height: 40px;
  font-size: 28px;
  margin: 20px;
}

<div>
  <input
    className="c0"
    data-testid="filter-name-input"
    onChange={[Function]}
    placeholder="Gnome Name"
    type="text"
    value="Dummy Name"
  />
</div>
`;
